clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/web/DynamicDataClient.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.clients.simpleclient.web;
+
+import feign.Response;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import uk.gov.gchq.palisade.clients.simpleclient.request.AddSerialiserRequest;
+import uk.gov.gchq.palisade.clients.simpleclient.request.ReadRequest;
+
+/**
+ * The interface Dynamic data client.
+ */
+public interface DynamicDataClient {
+    /**
+     * The interface Data client.
+     */
+    interface DataClient {
+
+        /**
+         * Read chunked response.
+         *
+         * @param request the request
+         * @return the response
+         */
+        @PostMapping(value = "/read/chunked", consumes = "application/json", produces = "application/octet-stream")
+        Response readChunked(@RequestBody final ReadRequest request);
+
+        /**
+         * Add serialiser boolean.
+         *
+         * @param request the request
+         * @return the boolean
+         */
+        @PostMapping(value = "/addSerialiser", consumes = "application/json", produces = "application/json")
+        Boolean addSerialiser(@RequestBody final AddSerialiserRequest request);
+
+    }
+
+    /**
+     * Client for data client.
+     *
+     * @param serviceId the service id
+     * @return the data client
+     */
+    DataClient clientFor(final String serviceId);
```

Comment:
What is the purpose of this method?

Reply:
so the urlDataClient and namedDataClient can implement it, and overwrite it with their FeignBuilders

Reply:
The problem being dynamically assigning values to the feign client name and url fields - ie. when a palisade response comes back, get the connection detail service name and create a new feign client with that name and lookup a url for that name from web.clients